src/xcode/ENA/ENASecurity/Sources/ENASecurity/CBCEncryption.swift

```diff
@@ -32,8 +39,20 @@ public struct CBCEncryption {
 
     private let encryptionKey: Data
     private let initializationVector: Data
+    private let ivLengthConstraint: Int?
 
     private func crypt(data: Data, option: CCOperation) -> Result<Data, CBCEncryptionError> {
+
+        guard !initializationVector.isEmpty else {
```

Comment:
🤔 wouldn't it be an idea to check for those constraints during `init` ?
Seems a little late to check it here.

Reply:
Since init does not have type safe error handling (only optional or throws), I would like to avoid failable initializers.